ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -706,10 +709,27 @@ private Page<NamedBlobRecord> run_list_v2(String accountName, String containerNa
         int resultIndex = 0;
         while (resultSet.next()) {
           String blobName = resultSet.getString(1);
-          if (resultIndex++ == maxKeysValue) {
+          if (resultIndex == maxKeysValue) {
             nextContinuationToken = blobName;
             break;
           }
+
+          if (groupDirectories) {
+            // Extract the portion after the prefix and before the next '/'
+            String remainingPath = blobName.substring(blobNamePrefix == null ? 0 : blobNamePrefix.length());
+            remainingPath = remainingPath.startsWith("/") ? remainingPath.substring(1) : remainingPath;
+            int delimiterIndex = remainingPath.indexOf("/");
+            if (delimiterIndex != -1) {
+              boolean validEntry = directories.add(remainingPath.substring(0, delimiterIndex) + "/");
```

Comment:
Nit: it could be remainingPath.substring(0, delimiterIndex +DELIMITER_STRING.length());

Reply:
Done